jgrapht-core/src/main/java/org/jgrapht/alg/scoring/BetweennessCentrality.java

```diff
@@ -0,0 +1,299 @@
+
+import java.util.*;
+
+import org.jgrapht.*;
+import org.jgrapht.alg.interfaces.*;
+import org.jgrapht.util.*;
+
+/**
+ * Betweenness centrality.
+ * 
+ * <p>
+ * Computes the betweenness centrality of each vertex of a graph. The betweenness centrality of a node
+ * $v$ is given by the expression: $g(v)= \sum_{s \neq v \neq
+ * t}\frac{\sigma_{st}(v)}{\sigma_{st}}$ where $\sigma_{st}$ is the total number of shortest paths
+ * from node $s$ to node $t$ and $\sigma_{st}(v)$ is the number of those paths that pass through
+ * $v$. For more details see
+ * <a href="https://en.wikipedia.org/wiki/Betweenness_centrality">wikipedia</a>.
+ * 
+ * The algorithm is based on 
+ * <ul>
+ * <li>randes, Ulrik (2001). "A faster algorithm for betweenness centrality". 
+ * Journal of Mathematical Sociology. 25 (2): 163–177.</li>
+ * </ul>
+ *
+ * The running time is $O(nm) and $O(nm +n^2 \log n)$ for unweighted and weighted graph respectively, 
+ * where $n$ is the number of vertices and $m$ the number of edges of the graph. The space
+ * complexity is $O(n + m)$.
+ *
+ * 
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ * 
+ * @author Assaf Mizrachi
+ * @since December 2017
+ */
+public class BetweennessCentrality<V, E>
+    implements VertexScoringAlgorithm<V, Double>
+{
+
+   /**
+     * Underlying graph
+     */
+    private final Graph<V, E> graph;
+    /**
+     * Whether to normalize scores
+     */
+    private final boolean normalize;
+    /**
+     * The actual scores
+     */
+    private Map<V, Double> scores;
+
+    /**
+     * Construct a new instance.
+     * 
+     * @param graph the input graph
+     */
+    public BetweennessCentrality(Graph<V, E> graph)
+    {
+        this(graph, false);
+    }
+    
+    /**
+     * Construct a new instance.
+     * 
+     * @param graph the input graph
+     * @param normalize whether to normalize by dividing the closeness by (n-1)*(n-2), where n is the
+     *        number of vertices of the graph
+     */
+    public BetweennessCentrality(Graph<V, E> graph, boolean normalize)
+    {
+        this.graph = Objects.requireNonNull(graph, "Graph cannot be null");
+
+        this.scores = null;
+        this.normalize = normalize;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Map<V, Double> getScores()
+    {
+        if (scores == null) {
+            compute();
+        }
+        return Collections.unmodifiableMap(scores);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Double getVertexScore(V v)
+    {
+        if (!graph.containsVertex(v)) {
+            throw new IllegalArgumentException("Cannot return score of unknown vertex");
+        }
+        if (scores == null) {
+            compute();
+        }
+        return scores.get(v);
+    }
+
+    /**
+     * Compute the centrality index
+     */
+    private void compute()
+    {
+        // initialize result container
+        initScores();

```

Comment:
Consider inlining the method call.

Reply:
Not sure what you mean by "inlining" in this case. Can you explain/example?

Reply:
To put the method's body directly in the caller. See https://refactoring.com/catalog/inlineMethod.html .